webapp/manifest.json

```diff
@@ -45,7 +54,7 @@
 					]
 				},
 				{
-					"pattern": "product/{productId}",
+					"pattern": "product/{productId}/:mode:",
```

Comment:
Instead of it use search query options, so the link will look like
products/1?mode=create
products/1?mode=edit
products/1 (for readonly mode)